canonical-web-and-design/maas-ui
# feat(ui): teach MachineListTable to hide columns

## Done
Adds a new prop, `hideColumns` to `MachineListTable`, which takes an array of column names.
e.g. <MachineListTable hiddenColumns={['power', 'zone']} />

Note: the 'fqdn' column will not be hidden unless `showActions` is false.

I *haven't* migrated `MachineListTable` to ts here, as this is probably the largest component in the project, and should be tackled in a separate branch.

## QA

### MAAS deployment

To run this branch you will need access to one of the following MAAS deployments:

- [Karura](/HACKING.md#karura)
- [Bolla](/HACKING.md#bolla)
- [A development MAAS](/HACKING.md#development-deployment)
- [A local snap MAAS](/HACKING.md#snap-deployment) (this will not usually have machines)

### Running the branch

You can run this branch by:

- Serving with [dotrun](/HACKING.md#maas-ui-development-setup)
- [Building in a development MAAS](/HACKING.md#running-maas-ui-from-a-development-maas)

### QA steps

- Hack `MachineListTable`'s `hiddenColumns` prop, and add some column names, e.g.

`hiddenColumns={['power', 'zone']}`

- Visit the machine listing, and both specified columns and headers should be hidden.

## Fixes

Fixes: canonical-web-and-design/MAAS-squad#2074

## Launchpad issue

Related Launchpad maas issue in the form `lp#number`.

## Backports

In general, please propose fixes against *master* rather than release branches (e.g. 2.7), unless the fix is only applicable for that specific release. Please apply backport labels to the PR (e.g. "Backport 2.7") for the appropriate releases to target.

Only bug and security fixes should be backported, new features should only land in master.
